tasks/storage_deal.go

```diff
@@ -379,7 +384,14 @@ func (de *storageDealExecutor) executeAndMonitorDeal(ctx context.Context, update
 			}
 
 			if len(info.DealStages.Stages) > 0 {
-				err = updateStage(ctx, storagemarket.DealStates[info.State], toStageDetails(info.DealStages.Stages[len(info.DealStages.Stages)-1]))
+				newState := storagemarket.DealStates[info.State]
```

Comment:
This also it not going to work. We don't get the DealID till the deal is on chain unfortunately, so we have to wait till at least the publish point to get the deal ID